src/main/java/Zap.java

```diff
@@ -0,0 +1,236 @@
+        System.out.println("   [E][ ] " + description + " (From: " + startTime + " To: " + endTime + ")");
+        if (tasks.size() == 1) {
+            System.out.println(" Now you have 1 task in the list.");
+        } else {
+            System.out.println(" Now you have " + tasks.size() + " tasks in the list.");
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * 'displayTasks' method displays all the lists -- marked and unmarked
+     * when the command 'list' is inputted by the user
+     */
+    private static void displayTasks() {
+        if (tasks.isEmpty()) {
+            System.out.println("____________________________________________________________");
+            System.out.println(" Tasks list is empty.");
+            System.out.println("____________________________________________________________");
+        } else {
+            System.out.println("____________________________________________________________");
+            System.out.println(" Here are the tasks in your list:");
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println(" " + (i + 1) + "." + tasks.get(i));
+            }
+            System.out.println("____________________________________________________________");
+        }
+    }
+
+    /**
+     * `markTask` method marks a task as done.
+     * it extracts the task index, and updates the task's status
+     */
+    private static void markTask(String userCommand) {
+        String[] characters = userCommand.split("\\s+");
+
+        if (characters.length != 2 || !characters[1].matches("\\d+")) {
+            System.out.println("Invalid command format. Please use 'mark <task index>'.");
+            return;
+        }
+
+        int taskIndex = Integer.parseInt(characters[1]) - 1;
+
+        if (isValidTaskIndex(taskIndex)) {
+            tasks.get(taskIndex).markAsDone();
+            System.out.println("____________________________________________________________");
+            System.out.println(" Nice! I've marked this task as done:");
+            System.out.println("   " + tasks.get(taskIndex));
+            System.out.println("____________________________________________________________");
+        } else {
+            System.out.println("Invalid task index. Please enter a valid number.");
+        }
+    }
+
+    /**
+     * `unmarkTask` method unmarks a task that was originally marked as done.
+     * it extracts the task index, and updates the task's status, then prints a confirmation message.
+     */
+    private static void unmarkTask(String userCommand) {
+        String[] characters = userCommand.split("\\s+");
+
+        if (characters.length == 2 && characters[1].matches("\\d+")) {
+            int taskIndex = Integer.parseInt(characters[1]) - 1;
+
+            if (isValidTaskIndex(taskIndex)) {
+                tasks.get(taskIndex).markAsNotDone();
```

Comment:
seems wordy with lots of criteria to meet. Would be good to create functions for these lengthy criteria